File: commons/dto/src/main/java/net/blueshell/dto/SimpleCommitteeDTO.java
package net.blueshell.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class SimpleCommitteeDTO extends BaseDTO {

    @JsonProperty("id")
    private Long id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("description")
    private String description;
}

File: api/src/main/kotlin/net/blueshell/api/domain/committee/web/dto/SimpleCommitteeDTO.kt
package net.blueshell.api.domain.committee.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size
import net.blueshell.api.shared.dto.AuditedAutoIdDTO

@Schema(name = "SimpleCommittee")
data class SimpleCommitteeDTO(
    @field:NotBlank
    @field:Size(max = 255)
    var name: String? = null,

    @field:NotBlank
    @field:Size(max = 4095)
    var description: String? = null,
) : AuditedAutoIdDTO()
